src/synapse/workload.py

```diff
@@ -678,6 +678,36 @@ def enable_smtp(container: ops.Container, charm_state: CharmState) -> None:
         raise EnableSMTPError(str(exc)) from exc
 
 
+def enable_redis(container: ops.Container, charm_state: CharmState) -> None:
+    """Change the Synapse configuration to enable Redis.
+
+    Args:
+        container: Container of the charm.
+        charm_state: Instance of CharmState.
+
+    Raises:
+        WorkloadError: something went wrong enabling SMTP.
+    """
+    try:
+        config = container.pull(SYNAPSE_CONFIG_PATH).read()
+        current_yaml = yaml.safe_load(config)
+        current_yaml["redis"] = {}
+
+        if charm_state.redis_config is None:
+            raise WorkloadError(
+                "Redis Configuration not found. "
+                "Please verify the integration between Redis and Synapse."
+            )
+
+        redis_config = charm_state.redis_config
+        current_yaml["redis"]["enabled"] = True
+        current_yaml["redis"]["host"] = redis_config["host"]
+        current_yaml["redis"]["port"] = redis_config["port"]
```

Comment:
This means that we have two locations storing the same state. This is not good because we now need to make sure that they are in sync.  
Another limitation for the central truth of a charmState...

Reply:
My bad, misunderstood this one